src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -587,16 +587,15 @@ static int add_partition_esp(DissectedPartition *p, bool has_xbootldr) {
          * Otherwise, we do nothing. */
         if (!has_xbootldr && slash_boot_exists()) {
                 r = slash_boot_in_fstab();
+                if (r < 0)
+                        return r;
```

Comment:
This checks if `/boot` is in fstab, and return true if it exists. Ignoring non-negative value is meaningless, and if this really fixes the issue, then please drop the call entirely.